src/graphapi/find_engine.cpp

```diff
@@ -36,12 +36,163 @@
 #include <miopen/graphapi/rng.hpp>
 #include <miopen/graphapi/util.hpp>
 #include <miopen/graphapi/variant_pack.hpp>
+#include <miopen/graphapi/convolution.hpp>
+#include <miopen/graphapi/conv_bias_res_add_activ_forward_executor.hpp>
 
 namespace miopen {
 namespace graphapi {
 
 GraphPatternMatcher::~GraphPatternMatcher() = default;
 
+class ConvBiasResAddActive_Fwd_Pattern : public GraphPatternMatcher
+{
+    static const OpGraph& getPatternGraph()
+    {
+        static auto graph_gen =
+            PatternGraphGenerator::Make({{"OP_CONVOLUTION_FORWARD", {"X", "W"}, {"T_C_0"}},
+                                         {"OP_POINTWISE:ADD", {"T_C_0", "Z"}, {"T_A_0"}},
+                                         {"OP_POINTWISE:ADD", {"T_A_0", "BIAS"}, {"T_A_1"}},
+                                         {"OP_POINTWISE:RELU_FWD", {"T_A_1"}, {"Y"}}});
+
+        return graph_gen->graph();
+    }
+
+    static bool isBiasNode(OperationPointwise* addNode)
+    {
+        auto& lengthsToCheck  = addNode->getX()->isVirtual() ? addNode->getB()->GetLengths()
+                                                             : addNode->getX()->GetLengths();
+        int notOneOrZeroCount = 0;
+        for(auto& length : lengthsToCheck)
+        {
+            if(length > size_t{1})
+            {
+                notOneOrZeroCount++;
+            }
+        }
+
+        // Assume bias node when at most one dimension is larger than 1.
+        return notOneOrZeroCount <= 1;
+    }
+
+    static bool hasBiasNode(const OpGraph& graph)
+    {
+        auto* conv = dynamic_cast<OperationConvolutionForward*>(
+            graph.findOutNeighByName(graph.getSourceNode(), "OP_CONVOLUTION_FORWARD"));
+        assert(conv);
+
+        auto* add1 =
+            dynamic_cast<OperationPointwise*>(graph.findOutNeighByName(conv, "OP_POINTWISE:ADD"));
+        assert(add1);
+
+        auto* add2 =
+            dynamic_cast<OperationPointwise*>(graph.findOutNeighByName(add1, "OP_POINTWISE:ADD"));
+        assert(add2);
+
+        return isBiasNode(add1) || isBiasNode(add2);
+    }
+
+public:
+    static std::unique_ptr<GraphPatternMatcher> Make()
+    {
+        return std::make_unique<ConvBiasResAddActive_Fwd_Pattern>();
+    }
+
+    std::string_view name() const final
+    {
+        static const char* n = "convbiasresaddactivation_fwd";
+        return n;
+    }
+
+    bool matches(const OpGraph* graph_ptr) const final
+    {
+        assert(graph_ptr);
+
+        if(!isIsomorphic(*graph_ptr, getPatternGraph()))
+        {
+            return false;
+        }
+
+        return hasBiasNode(*graph_ptr);
+    }
+
+    std::vector<Engine> getEngines(OpGraph* graph_ptr) const override
+    {
+        assert(graph_ptr);
+        assert(matches(graph_ptr));
```

Comment:
I think in this case it will throw if it doesn't match, (since the code below will fail to find, and explode), and we don't want to pay the runtime cost of checking matches again. I was following how the other pattern matcher was set up, and I think that is the reason anyways.